### PR TITLE
[Refactor] refactor native analyse to support external analyze

### PR DESCRIPTION
**Pull request by zombee0**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 122 / 199 (61.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeJob.java|0|1|00.00%|[145]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|7|00.00%|[161, 162, 501, 502, 526, 527, 528]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|2|00.00%|[689, 694]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJob.java|4|18|22.22%|[161, 162, 165, 166, 167, 169, 171, 172, 173, 174, 178, 179, 180, 181]|
|:large_blue_circle:|com/starrocks/sql/ast/DropStatsStmt.java|1|4|25.00%|[39, 43, 44]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|2|6|33.33%|[843, 848, 1596, 1597]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|9|21|42.86%|[879, 933, 934, 936, 957, 958, 961, 962, 964, 965, 966, 967]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|7|15|46.67%|[112, 113, 114, 116, 129, 199, 200, 253]|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeStmt.java|2|4|50.00%|[89, 90]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|4|7|57.14%|[246, 258, 259]|
|:large_blue_circle:|com/starrocks/statistic/NativeAnalyzeStatus.java|57|74|77.03%|[98, 102, 114, 123, 127, 201, 208, 212, 213, 214, 215, 221, 223, 231, 232, 250, 251]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|12|14|85.71%|[102, 107]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|11|12|91.67%|[408]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeStatus.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java [hunk 1]
```diff
@@ -113,8 +123,14 @@ public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, Conne
             if (null != statement.getTableName()) {
                 TableName tbl = statement.getTableName();
 
+                if ((Strings.isNullOrEmpty(tbl.getCatalog()) &&
+                        CatalogMgr.isExternalCatalog(session.getCurrentCatalog())) ||
+                        CatalogMgr.isExternalCatalog(tbl.getCatalog())) {
+                    throw new SemanticException("External Table don't support analyze job");
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:129]: Why not support analyze job?

File: fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java [hunk 2]
```diff
@@ -1586,11 +1587,15 @@ public void logRemoveAnalyzeJob(AnalyzeJob job) {
     }
 
     public void logAddAnalyzeStatus(AnalyzeStatus status) {
-        logEdit(OperationType.OP_ADD_ANALYZE_STATUS, status);
+        if (status instanceof NativeAnalyzeStatus) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java:1590]: AnalyzeStatus.isNative()

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java [hunk 3]
```diff
@@ -240,15 +240,26 @@ public AnalyzeStatus collectStatistics(ConnectContext statsConnectCtx,
...
             GlobalStateMgr.getCurrentAnalyzeMgr().unregisterConnection(analyzeStatus.getId(), false);
         }
 
         analyzeStatus.setStatus(StatsConstants.ScheduleStatus.FINISH);
         analyzeStatus.setEndTime(LocalDateTime.now());
-        GlobalStateMgr.getCurrentAnalyzeMgr().addAnalyzeStatus(analyzeStatus);
+        if (analyzeStatus instanceof NativeAnalyzeStatus) {
+            GlobalStateMgr.getCurrentAnalyzeMgr().addAnalyzeStatus(analyzeStatus);
+        } else {
+            GlobalStateMgr.getCurrentAnalyzeMgr().replayAddAnalyzeStatus(analyzeStatus);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java:258]: why replay here?

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java:258]: will do it

File: fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java (hunk 2 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java:1590]: will do it

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java (hunk 3 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticExecutor.java:258]: This interface is no longer the meaning of replay, is it more appropriate to change it to update?

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 123 / 202 (60.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeJob.java|0|1|00.00%|[145]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|9|00.00%|[137, 138, 165, 166, 505, 506, 530, 531, 532]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|2|00.00%|[689, 694]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJob.java|4|18|22.22%|[161, 162, 165, 166, 167, 169, 171, 172, 173, 174, 178, 179, 180, 181]|
|:large_blue_circle:|com/starrocks/sql/ast/DropStatsStmt.java|1|4|25.00%|[39, 43, 44]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|2|6|33.33%|[843, 848, 1596, 1597]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|9|21|42.86%|[879, 933, 934, 936, 957, 958, 961, 962, 964, 965, 966, 967]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|7|15|46.67%|[112, 113, 114, 116, 129, 199, 200, 253]|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeStmt.java|2|4|50.00%|[89, 90]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|4|7|57.14%|[246, 258, 259]|
|:large_blue_circle:|com/starrocks/statistic/NativeAnalyzeStatus.java|58|75|77.33%|[103, 107, 119, 128, 132, 206, 213, 217, 218, 219, 220, 226, 228, 236, 237, 255, 256]|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|12|14|85.71%|[102, 107]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|11|12|91.67%|[409]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeStatus.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|1|1|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**